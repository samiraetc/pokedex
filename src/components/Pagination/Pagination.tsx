import React from 'react';
import { FiArrowLeft, FiArrowRight } from "react-icons/fi";

export const COMPONENT_ID = 'pokedex-pagination'

type Pagination = {
    handleNextPage: any;
    handleBackPage: any;
    page: number,
    totalPages: number,
}

const Pagination = ({ handleBackPage, handleNextPage, page, totalPages }: Pagination) => {

    return (
        <div className="pagination-container" id={COMPONENT_ID}>
        <button onClick={handleBackPage} className="pagination-button" id={`${COMPONENT_ID}-back`}><div><FiArrowLeft size={30} color="#373E59"/></div></button>
        <div className='pagination-page'>{page} of {totalPages}</div>
        <button onClick={handleNextPage} className="pagination-button" id={`${COMPONENT_ID}-next`}><div><FiArrowRight size={30} color="#373E59"/></div></button>
    </div>
    )
}

export default Pagination;