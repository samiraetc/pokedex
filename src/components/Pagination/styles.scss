.pagination-container {
    display: flex;
    flex-direction: row;
    text-align: center;
    gap: $pokedex-spacing-12;

    .pagination-button {
        background-color: $pokedex-color-white;
        border: none;
    }

    .pagination-page {
        color: $pokedex-color-blue-strong;
        font-weight: $pokedex-font-weight-bold;
    }

}