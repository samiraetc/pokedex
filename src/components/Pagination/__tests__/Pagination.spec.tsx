import { render, configure, fireEvent } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';
import Pagination, { COMPONENT_ID } from '../Pagination';

configure({
    testIdAttribute: 'id',
  });
  
describe('[PAGINATION]', () => {
    const Mock = jest.fn();
    test('should render Pagination', () => {
        const { asFragment, getByText, getByTestId } = render(<Pagination handleBackPage={Mock} handleNextPage={Mock}  page={1} totalPages={10}/>);

        expect(getByText('1 of 10'));
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

});