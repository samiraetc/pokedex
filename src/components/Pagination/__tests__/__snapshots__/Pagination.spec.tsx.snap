// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PAGINATION] should render Pagination 1`] = `
<DocumentFragment>
  <div
    class="pagination-container"
    id="pokedex-pagination"
  >
    <button
      class="pagination-button"
      id="pokedex-pagination-back"
    >
      <div>
        <svg
          color="#373E59"
          fill="none"
          height="30"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(55, 62, 89);"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="5"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 19 5 12 12 5"
          />
        </svg>
      </div>
    </button>
    <div
      class="pagination-page"
    >
      1 of 10
    </div>
    <button
      class="pagination-button"
      id="pokedex-pagination-next"
    >
      <div>
        <svg
          color="#373E59"
          fill="none"
          height="30"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(55, 62, 89);"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="5"
            x2="19"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 5 19 12 12 19"
          />
        </svg>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
